pages/gallery/gallery.js

```diff
@@ -0,0 +1,420 @@
+        @click:append-outer="startSearch"
+        @click:clear="clearSearch"
+      ></v-text-field>
+    </div>
+    
+    <v-list
+      two-line
+      style="padding: 10px;"
+    >
+      <template v-if="loading.status == true">
+        <div class="text-h6 py-6" style="padding: 0 10px !important">{{ loading.comments.length }}/{{ loading.steps }} {{ $root.lang().gallery.loading_message.general }}</div>
+        <span style="padding: 0 10px;">{{ loading.comments[loading.comments.length - 1] }}</span>
+      </template>
+      <template v-if="!loading.status && displayedTextures.length === 0">
+        <div class="text-h6 py-6" style="padding: 0 10px 10px !important">{{ $root.lang().global.no_results }}</div>
+      </template>
+
+      <div
+        v-if="!loading.status"
+        class="gallery-textures-container"
+      >
+        <div
+          v-for="(texture, index) in displayedTextures"
+          v-if="index <= displayedResults"
+          class="gallery-texture-in-container"
+          v-tooltip.right-start="{content: () => getAuthor(texture.textureID), html: true}"
+          v-on:click="openModal(texture.textureID)"
+        >
+          <img class="gallery-texture-image" onerror="this.onerror=null;this.src='https://database.compliancepack.net/images/bot/error.png';" :src="getTextureURL(texture.useID)" lazy-src="https://database.compliancepack.net/images/bot/loading.gif" />
+          <v-img class="gallery-texture-background" contain style="position: absolute; z-index: -1;" src="https://raw.githubusercontent.com/Compliance-Resource-Pack/Website/master/image/background/transparency_16x.png" />
+        </div>
+      </div>
+    </v-list>
+
+    <texture-modal
+      :opened="modalOpen"
+      :closeModal="closeModal"
+      :textureID="modalTextureID"
+      :textureObj="modalTextureObj"
+      :contributors="displayed.contributors"
+    ></texture-modal>
+
+    <v-btn icon large @click="toTop" v-show="scrollY > 300" class="go_up_btn">
+      <v-icon>
+        mdi-arrow-up
+      </v-icon>
+    </v-btn>
+  </v-container>
+  `,
+  data() {
+    return {
+      loading: {
+        status: false,
+        steps: 0,
+        comments: []
+      },
+      options: {
+        resolutions: ['16x', ...settings.resolutions],
+        tags: ['All'],
+        versions: settings.versions.java,
+        editions: settings.editions.map(e => e.toLowerCase())
+      },
+      current: {
+        resolution: settings.resolutions[0],
+        tag: 'All',
+        version: settings.versions.java[0],
+        edition: settings.editions[0],
+        search: null
+      },
+      dataJSON: {},
+      displayed: {
+        paths: {},
+        uses: {},
+        textures: {},
+        contributions: {
+          c32: [],
+          c64: []
+        },
+        contributors: {}
+      },
+      displayedResults: 20,
+      displayedTextures: [],
+      modalTextureID: null,
+      modalTextureObj: {},
+      modalOpen: false,
+    }
```

Comment:
A lot of data, sure you don't want to split this file in smaller or reusable components?

Reply:
I'm not sure we understand each other there ❓

Reply:
Do you need all this data ? Can't you split it in other components with custom v-model ?

Reply:
It's easier like that